OpenRA.Game/Graphics/HardwarePalette.cs

```diff
@@ -60,6 +61,14 @@ public void AddPalette(string name, ImmutablePalette p, bool allowModifiers)
 			int index = palettes.Count;
 			indices.Add(name, index);
 			palettes.Add(name, p);
+
+			if (palettes.Count > buffer.GetLength(0))
+			{
+				var oldBuffer = buffer;
+				buffer = new uint[Exts.NextPowerOf2(palettes.Count), Palette.Size];
```

Comment:
Somebody would need to update the ```Texture.SetData(uint[,])``` method and fix all callers. I'm ambivalent to such a change, but it's a job for a new PR IMO.